### PR TITLE
add missing container security context settings and tests

### LINKED ISSUE #2353: [Bug]: Missing SecurityContext in several Job Container spec
### What happened?

I am using jaeger-operator on k8s 1.24 and I'm planning to update to 1.25.
From security perspective my namespaces are using restricted PSS mode. 
I added ContainerSecurityContext settings to my CustomResource. The operator fails to create Jobs because ContainerSecurityContext is missing.

It happens, because SecurityContext is not set for containers here:
- https://github.com/jaegertracing/jaeger-operator/blob/50cd8c138b6f0b7de6463097069aa6f024b7e0bb/pkg/storage/elasticsearch_dependencies.go#L56
- https://github.com/jaegertracing/jaeger-operator/blob/50cd8c138b6f0b7de6463097069aa6f024b7e0bb/pkg/storage/cassandra_dependencies.go#L195
- https://github.com/jaegertracing/jaeger-operator/blob/50cd8c138b6f0b7de6463097069aa6f024b7e0bb/pkg/cronjob/spark_dependencies.go#L90
- https://github.com/jaegertracing/jaeger-operator/blob/50cd8c138b6f0b7de6463097069aa6f024b7e0bb/pkg/cronjob/es_rollover.go#L127

### Steps to reproduce

Enable restricted PSS profile on the namespace within jaeger-operator should create a Jaeger instance.
```
pod-security.kubernetes.io/enforce: restricted
pod-security.kubernetes.io/enforce-version: latest
```

### Expected behavior

Jaeger pods, jobs and cronjobs are created.

### Relevant log output

```shell
Warning  FailedCreate  2m39s  job-controller  Error creating: pods "jaeger-es-rollover-create-mapping-4gttf" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "jaeger-es-rollover-create-mapping" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "jaeger-es-rollover-create-mapping" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "jaeger-es-rollover-create-mapping" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "jaeger-es-rollover-create-mapping" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

### Screenshot

_No response_

### Additional context

_No response_

### Jaeger backend version

v1.49.0

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

_No response_

### Operating system

_No response_

### Deployment model

Kubernetes

### Deployment configs

_No response_

### PR DESCRIPTION
**Pull request by tingeltangelthomas**
## Which problem is this PR solving?
- Resolves #2353

## Description of the changes
- This PR adds container securityContext settings to several kubernetes templates

## How was this change tested?
- added and ran unittests for each change
- installed operator to my EKS cluster and created Jaeger instance in a restricted namespace
- executed `make lint test`

## Checklist
- [X] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [X] I have signed all commits
- [X] I have added unit tests for the new functionality
- [X] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by tingeltangelthomas** [comment]
Hi @iblancasa,
is there anything I can do to make the tests green?

/retest-required

**Comment by iblancasa** [comment]
@rubenvp8510 can you take a look to this PR?

File: pkg/cronjob/es_rollover.go [hunk 1]
```diff
@@ -125,13 +125,14 @@ func createTemplate(name, action string, jaeger *v1.Jaeger, envs []corev1.EnvVar
...
-					Resources:    commonSpec.Resources,
-					VolumeMounts: commonSpec.VolumeMounts,
+					Name:            name,
+					Image:           util.ImageName(jaeger.Spec.Storage.EsRollover.Image, "jaeger-es-rollover-image"),
+					Args:            []string{action, util.GetEsHostname(jaeger.Spec.Storage.Options.Map())},
+					Env:             util.RemoveEmptyVars(envs),
+					EnvFrom:         envFromSource,
+					Resources:       commonSpec.Resources,
+					VolumeMounts:    commonSpec.VolumeMounts,
+					SecurityContext: commonSpec.ContainerSecurityContext,
```
Comment (rubenvp8510) [inline pkg/cronjob/es_rollover.go:135]: Does this have a default value? what happens if it's not set in the CR?

Comment (tingeltangelthomas) [reply pkg/cronjob/es_rollover.go:135]: Hi @rubenvp8510 , 
securityContext will be `{}` (empty)
The same here: https://github.com/jaegertracing/jaeger-operator/blob/2b0af44fc671da6d6eb0126e8eb78999beb09df6/pkg/deployment/agent.go#L196C60-L196C60

Comment (rubenvp8510) [reply pkg/cronjob/es_rollover.go:135]: Make sense! Thanks!

**Comment by iblancasa** [comment]
/restest

I think there si something going wrong in OpenShift with these changes.

**Comment by rubenvp8510** [comment]
> /restest
> 
> I think there si something going wrong in OpenShift with these changes.

Yeah :/ or may be is just matter of update the E2Es,

**Comment by tingeltangelthomas** [comment]
I'm a bit disappointed about openshift e2e tests.
Can we check if failures are related to my changes? Do I need to revert my changes and retest or is there an easier way?

Thank you!

**Review by iblancasa — APPROVED** [review]

**Pull request merged**